src/ImageProcessor.Web/HttpModules/ImageProcessingModule.cs

```diff
@@ -826,7 +826,10 @@ private IImageService GetImageServiceForRequest(string url, string applicationPa
             foreach (IImageService service in services)
             {
                 string key = service.Prefix;
-                if (!string.IsNullOrWhiteSpace(key) && path.StartsWith(key, StringComparison.InvariantCultureIgnoreCase))
+                if (!string.IsNullOrWhiteSpace(key) 
+                    && (service.IsRegex 
```

Comment:
After I get [ImageSharp](https://github.com/JimBobSquarePants/ImageSharp) V1 out of the way most likely.  Could be as late as the end of the year.